tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,1138 @@
+                == "application/json"
+            )
+            assert (
+                response["DriftCheckBaselines"]["ModelDataQuality"]["Statistics"]["ContentType"]
+                == "application/json"
+            )
+            assert response["CustomerMetadataProperties"] == customer_metadata_properties
+            break
+    finally:
+        try:
+            pipeline.delete()
+        except Exception as error:
+            logging.error(error)
+
+
+def test_model_registration_with_model_repack(
+    pipeline_session,
+    role,
+    pipeline_name,
+):
+    kms_key = get_or_create_kms_key(pipeline_session, role)
+    base_dir = os.path.join(DATA_DIR, "pytorch_mnist")
+    entry_point = os.path.join(base_dir, "mnist.py")
+    input_path = pipeline_session.upload_data(
+        path=os.path.join(base_dir, "training"),
+        key_prefix="integ-test-data/pytorch_mnist/training",
+    )
+    inputs = TrainingInput(s3_data=input_path)
+
+    instance_count = ParameterInteger(name="InstanceCount", default_value=1)
+    instance_type = ParameterString(name="InstanceType", default_value="ml.m5.xlarge")
+    good_enough_input = ParameterInteger(name="GoodEnoughInput", default_value=1)
+
+    pytorch_estimator = PyTorch(
+        entry_point=entry_point,
+        role=role,
+        framework_version="1.5.0",
+        py_version="py3",
+        instance_count=instance_count,
+        instance_type=instance_type,
+        sagemaker_session=pipeline_session,
+        output_kms_key=kms_key,
+    )
+    train_step_args = pytorch_estimator.fit(inputs=inputs)
+    step_train = TrainingStep(
+        name="pytorch-train",
+        step_args=train_step_args,
+    )
+    model = Model(
+        name="MyModel",
+        image_uri=pytorch_estimator.training_image_uri(),
+        model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
+        sagemaker_session=pipeline_session,
+        role=role,
+        entry_point=entry_point,
+        source_dir=base_dir,
+        model_kms_key=kms_key,
+    )
+    # register model with runtime repack
+    regis_model_step_args = model.register(
+        content_types=["text/csv"],
+        response_types=["text/csv"],
+        inference_instances=["ml.t2.medium", "ml.m5.large"],
+        transform_instances=["ml.m5.large"],
+        description="test-description",
+        model_package_group_name=f"{pipeline_name}TestModelPackageGroup",
+    )
+    step_model_regis = ModelStep(
+        name="pytorch-register-model",
+        step_args=regis_model_step_args,
+    )
+
+    # create model without runtime repack
+    model.entry_point = None
+    model.source_dir = None
+    create_model_step_args = model.create(
+        instance_type="ml.m5.large",
+        accelerator_type="ml.eia1.medium",
+    )
+
+    step_model_create = ModelStep(
+        name="pytorch-model",
+        step_args=create_model_step_args,
+    )
+
+    step_cond = ConditionStep(
```

Comment:
Remove the condition step. We're not testing the condition steps here.

Reply:
Removed in the new commit